cab-token-generator/javatests/com/google/auth/credentialaccessboundary/ClientSideCredentialAccessBoundaryFactoryTest.java

```diff
@@ -745,6 +745,9 @@ public void generateToken_withAvailablityCondition_success() throws Exception {
     CabToken cabToken = parseCabToken(token);
     assertEquals("accessToken", cabToken.intermediateToken);
 
+    // Verifies the encrypted restriction has no padding
+    assertFalse(cabToken.encryptedRestriction.contains(String.valueOf("=")));
```

Comment:
qq, does the Base64.Encoder keep the `=` by default? I was assuming that it was keeping something like additional spaces at the end. Would the test case make more sense to ensure that the token value doesn't end with spaces? If so, do you have an example/ mock token that runs into these padding issues?

Reply:
`=` is used to [pad the base64 encoding output](https://en.wikipedia.org/wiki/Base64#Output_padding) if the input length in bytes is not a multiple of 3. This is the default behavior for Base64.Encoder if `withoutPadding` is not used. The base64 encoded restriction of this test case includes the padding at the end.

Reply:
Gotcha, thanks! Can you just add a small comment above in the test to explain that padding adds `=` for values not in multiple of 3 (just for future maintainers who don't know this/ aren't familiar with padding)

Reply:
Added.